src/threads/model.rs

```diff
@@ -135,14 +137,175 @@ impl Threads {
     }
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ThreadCommentThreadPayload {
+    thread_id: i32,
+    content: String,
+    user_id: i32,
+}
+
+#[derive(Error, Debug)]
+pub enum ThreadCommentThreadError {
+    #[error("Comment could not be saved because of a database error: {inner:?}.")]
+    DatabaseError { inner: sqlx::Error },
+    #[error("Comment could not be saved because thread is archived")]
+    ThreadArchivedError,
+    #[error("Comment could not be saved because of a database error: {inner:?}.")]
+    EventError { inner: EventError },
+}
+
+impl From<sqlx::Error> for ThreadCommentThreadError {
+    fn from(inner: sqlx::Error) -> Self {
+        ThreadCommentThreadError::DatabaseError { inner }
+    }
+}
+
+impl From<EventError> for ThreadCommentThreadError {
+    fn from(error: EventError) -> Self {
+        match error {
+            EventError::DatabaseError { inner } => inner.into(),
+            inner => ThreadCommentThreadError::EventError { inner },
+        }
+    }
+}
+
+impl Threads {
+    pub async fn comment_thread<'a, E>(
+        payload: ThreadCommentThreadPayload,
+        executor: E,
+    ) -> Result<(), ThreadCommentThreadError>
+    where
+        E: Executor<'a>,
+    {
+        let mut transaction = executor.begin().await?;
+
+        let thread = sqlx::query!(
+            r#"
+                SELECT instance_id, archived
+                    FROM comment
+                    WHERE id = ?
+            "#,
+            payload.thread_id
+        )
+        .fetch_one(&mut transaction)
+        .await?;
+
+        if thread.archived != 0 {
+            return Err(ThreadCommentThreadError::ThreadArchivedError);
+        }
+
+        sqlx::query!(
+            r#"
+                INSERT INTO uuid (trashed, discriminator)
+                    VALUES (0, 'comment')
+            "#
+        )
+        .execute(&mut transaction)
+        .await?;
+
+        sqlx::query!(
+            r#"
+                INSERT INTO comment (id, date, archived, title, content, uuid_id, parent_id, author_id, instance_id )
+                    VALUES (LAST_INSERT_ID(), ?, 0, NULL, ?, NULL, ?, ?, ?)
+            "#,
+            DateTime::now(),
+            payload.content,
+            payload.thread_id,
+            payload.user_id,
+            thread.instance_id
+        )
+        .execute(&mut transaction)
+        .await?;
+
+        let value = sqlx::query!(r#"SELECT LAST_INSERT_ID() as id"#)
+            .fetch_one(&mut transaction)
+            .await?;
+        let comment_id = value.id as i32;
+
+        CreateCommentEventPayload::new(
+            payload.thread_id,
+            comment_id,
+            payload.user_id,
+            thread.instance_id,
+        )
+        .save(&mut transaction)
+        .await?;
+
+        let subscription = Subscription {
+            object_id: payload.thread_id,
+            user_id: payload.user_id,
+            send_email: true,
```

Comment:
@inyono: Review that. Although I wouldn't be surprised if that is the behavior of legacy ^^'

Reply:
Okay. This should actually be a param in the payload (if you want to subscribe and if you want to get emails).

Reply:
https://github.com/serlo/api.serlo.org/issues/234